utilities/tools/Set-ModuleReadMe.ps1

```diff
@@ -303,6 +303,81 @@ function Set-OutputsSection {
     return $updatedFileContent
 }
 
+<#
+.SYNOPSIS
+Add module references (cross-references) to the module's readme
+
+.DESCRIPTION
+Add module references (cross-references) to the module's readme. This includes both local (i.e., file path), as well as remote references (e.g., ACR)
+
+.PARAMETER TemplateFileContent
+Mandatory. The template file content object to crawl data from
+
+.PARAMETER ReadMeFileContent
+Mandatory. The readme file content array to update
+
+.PARAMETER SectionStartIdentifier
+Optional. The identifier of the 'outputs' section. Defaults to '## Cross-referenced modules'
+
+.EXAMPLE
+Set-ReferencesSection -TemplateFileContent @{ resource = @{}; ... } -ReadMeFileContent @('# Title', '', '## Section 1', ...)
+
+Update the given readme file's 'Cross-referenced modules' section based on the given template file content
+#>
+function Set-ReferencesSection {
+
+    [CmdletBinding(SupportsShouldProcess)]
+    param (
+        [Parameter(Mandatory)]
+        [hashtable] $TemplateFileContent,
+
+        [Parameter(Mandatory)]
+        [object[]] $ReadMeFileContent,
+
+        [Parameter(Mandatory = $false)]
+        [string] $SectionStartIdentifier = '## Cross-referenced modules'
+    )
+
+    . (Join-Path (Split-Path $PSScriptRoot -Parent) 'tools' 'Get-LinkedModuleList.ps1')
+
+    $moduleRoot = Split-Path $TemplateFilePath -Parent
+    $resourceTypeIdentifier = $moduleRoot.Replace('\', '/').Split('/modules/')[1].TrimStart('/')
+
+    # Process content
+    $SectionContent = [System.Collections.ArrayList]@(
+        'This section gives you an overview of all local-referenced module files (i.e., other CARML modules that are referenced in this module) and all remote-referenced files (i.e., Bicep modules that are referenced from a Bicep Registry or Template Specs).',
+        '',
+        '| Reference | Type |',
+        '| :-- | :-- |'
+    )
+
+    $dependencies = (Get-LinkedModuleList)[$resourceTypeIdentifier]
+
+    if ($dependencies.Keys -contains 'localPathReferences' -and $dependencies['localPathReferences']) {
+        foreach ($reference in ($dependencies['localPathReferences'] | Sort-Object)) {
+            $SectionContent += ("| ``{0}`` | {1} |" -f $reference, 'Local reference')
+        }
+    }
+
+    if ($dependencies.Keys -contains 'remoteReferences' -and $dependencies['remoteReferences']) {
+        foreach ($reference in ($dependencies['remoteReferences'] | Sort-Object)) {
+            $SectionContent += ("| ``{0}`` | {1} |" -f $reference, 'Remote reference')
```

Comment:
Does remote reference here allude to a module that is not in this library but exists in a template spec / bicep registry? If so, is this something the user needs to have in the documentation? As it does not technically require them to do any action in terms of 'NOT' deleting the local referenced module vs. the remote one?

Reply:
To your first question, yes, exactly.
Regarding the second question, see comment below.